src/OpenTelemetry.Instrumentation.AWS/Implementation/Metrics/AWSHistogram.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using Amazon.Runtime.Telemetry;
+using Amazon.Runtime.Telemetry.Metrics;
+
+namespace OpenTelemetry.Instrumentation.AWS.Implementation.Metrics;
+
+internal class AWSHistogram<T> : Histogram<T>
+    where T : struct
+{
+    private readonly System.Diagnostics.Metrics.Histogram<T> histogram;
+
+    public AWSHistogram(System.Diagnostics.Metrics.Histogram<T> histogram)
+    {
+        this.histogram = histogram;
+    }
+
+    public override void Record(T value, Attributes? attributes = null)
+    {
+        if (attributes != null)
+        {
+            this.histogram.Record(value, attributes.AllAttributes.ToArray());
```

Comment:
The array allocation of heap is not desirable.

Reply:
unfortunately I couldn't avoid that with the current APIs, `Histogram.Record` accepts params as an array, and the API that the AWS SDK exposes uses `IEnumerable` since the attributes can be mutable in other places, maybe at some point  `Histogram.Record` may accepts params as an  `IEnumerable`  which would be more flexible.

Reply:
`Histogram.Record` has many overloads to accepts attributes(Tags) without forcing allocation. 
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/metrics#metrics-api

It is unfortunate that the AWS wrapper is forcing heap allocation for metrics in hot path.

Reply:
@muhammad-othman When we move the SDK's observability changes into the V4 development branch we can add an `AttributesAsSpan` property and use it here to remove the allocation.

Reply:
Will create a v4 task for this, also this isn't the only place that we use `.ToArray()` so it should be useful to use it in other places including the tracing changes.

Reply:
can you add a TODO comment here that future aws sdk will offer api's to avoid allocation.?